src/components/AddressDetailExplorer.js

```diff
@@ -46,412 +49,362 @@ class AddressDetailExplorer extends React.Component {
-              // There were unfetched tokens
-              this.setState({ warnMissingTokens: total });
-            } else {
-              // This will turn off the missing tokens alert
-              this.setState({ warnMissingTokens: 0 });
-            }
+  const reloadTokenSummaryAndHistory = useCallback(
+    async (addressToReload, tokenToReload) => {
+      setLoadingSummary(true);
+      setLoadingHistory(true);
 
-            if (this.state.selectedToken && tokens[this.state.selectedToken]) {
-              // use has a selected token, we will keep the selected token
-              selectedToken = this.state.selectedToken;
-            } else {
-              const hathorUID = hathorLib.constants.NATIVE_TOKEN_UID;
-              if (tokens[hathorUID]) {
-                // If HTR is in the token list of this address, it's the default selection
-                selectedToken = hathorUID;
-              } else {
-                // Otherwise we get the first element, if there is one
-                const keys = Object.keys(tokens);
-                if (keys.length === 0) {
-                  // In case the length is 0, we have no transactions for this address
-                  this.setState({
-                    loadingTokens: false,
-                    loadingSummary: false,
-                    loadingHistory: false,
-                  });
-                  return;
-                }
-                selectedToken = keys[0];
-              }
-            }
+      try {
+        const balanceData = await addressApi.getBalance(addressToReload, tokenToReload);
+        if (!balanceData) {
+          // An error happened with the API call
+          setShowReloadTokenButton(true);
+          return undefined;
+        }
+        setBalance(balanceData);
 
-            const tokenDidChange =
-              selectedToken !== this.state.selectedToken || this.state.selectedToken === '';
-
-            this.setState(
-              {
-                addressTokens: tokens,
-                loadingTokens: false,
-                selectedToken,
-              },
-              () => {
-                if (tokenDidChange || !this.state.metadataLoaded) {
-                  this.getSelectedTokenMetadata(selectedToken);
-                }
-
-                // Update token in the URL
-                this.updateTokenURL(selectedToken);
-                this.reloadTokenData(selectedToken);
-              }
-            );
-          })
-          .catch(error => {
-            this.setState({
-              loadingTokens: false,
-              errorMessage: error.toString(),
-            });
-          });
+        await getHistoryData(addressToReload, tokenToReload);
+        await getSelectedTokenMetadata(tokenToReload);
+      } catch (error) {
+        setErrorMessage(error);
       }
-    );
-  };
+      setLoadingSummary(false);
+      return undefined;
+    },
+    [getHistoryData, getSelectedTokenMetadata]
```

Comment:
Both of these dependencies can be a simple function, if they are changed to be a funcion maybe this should too.

Reply:
`getHistoryData()` gets called independently of `getSelectedTokenMetadata()` within `onNextPageClicked()` and `onPreviousPageClicked()`.

I think that's why both these functions are separated: there's no need to make an additional metadata call on page change.